File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const countMap = {};

        nums.forEach((num) => {
            if(countMap[num] === undefined) countMap[num] = 0;

            countMap[num] = countMap[num] + 1;
        });

        return Object.entries(countMap)
        .sort((a, b) => b[1] - a[1])
        .map((entry) => entry[0])
        .slice(0, k);
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        countMap = {}

        for value in nums:
            if value in countMap:
                countMap[value] = countMap[value] + 1
                if(countMap[value] >= 2):
                    return True
            else: 
                countMap[value] = 1
        
        return False